nixos/modules/services/web-servers/apache-httpd/default.nix

```diff
@@ -110,25 +110,18 @@ let
   }
   ++ cfg.extraModules;
 
-  loggingConf = (
-    if cfg.logFormat != "none" then
-      ''
-        ErrorLog ${cfg.logDir}/error.log
-
-        LogLevel notice
+  loggingConf = ''
+    ErrorLog ${if cfg.logFormat != "none" then "${cfg.logDir}/error.log" else "/dev/null"}
+    ${optionalString (cfg.logLevel != null) "LogLevel ${cfg.logLevel}"}
 
-        LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
-        LogFormat "%h %l %u %t \"%r\" %>s %b" common
-        LogFormat "%{Referer}i -> %U" referer
-        LogFormat "%{User-agent}i" agent
+    LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
+    LogFormat "%h %l %u %t \"%r\" %>s %b" common
+    LogFormat "%{Referer}i -> %U" referer
+    LogFormat "%{User-agent}i" agent
+    ${optionalString (cfg.logFormat == "custom") "LogFormat \"${cfg.comstomLogFormat}\" custom"}
```

Comment:
This needs some sort of escaping. As it is now if I set 

```nix
customLogFormat = ''The X-Forwarded-For header was "%{X-Forwarded-For}", and here's some other stuff in quotes: "%l %u %t"'';
```

it will produce a syntactically invalid config.

Sorry I didn't catch this on the first review.